homeassistant/components/light/template.py

```diff
@@ -0,0 +1,192 @@
+"""
+Support for Template lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.template/
+"""
+import logging
+import asyncio
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ENTITY_ID_FORMAT, Light, SUPPORT_BRIGHTNESS)
+from homeassistant.const import (
+    CONF_VALUE_TEMPLATE, ATTR_ENTITY_ID, ATTR_FRIENDLY_NAME, STATE_ON,
+    STATE_OFF, EVENT_HOMEASSISTANT_START
+)
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.exceptions import TemplateError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.restore_state import async_get_last_state
+from homeassistant.helpers.script import Script
+
+_LOGGER = logging.getLogger(__name__)
+_VALID_STATES = [STATE_ON, STATE_OFF, 'true', 'false']
+
+CONF_LIGHTS = 'lights'
+ON_ACTION = 'turn_on'
+OFF_ACTION = 'turn_off'
+LEVEL_ACTION = 'set_level'
+LEVEL_TEMPLATE = 'level_template'
+
+
+LIGHT_SCHEMA = vol.Schema({
+    vol.Required(ON_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Required(OFF_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Optional(CONF_VALUE_TEMPLATE, default=None): cv.template,
+    vol.Optional(LEVEL_ACTION, default=None): cv.SCRIPT_SCHEMA,
+    vol.Optional(LEVEL_TEMPLATE, default=None): cv.template,
+    vol.Optional(ATTR_FRIENDLY_NAME, default=None): cv.string,
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids
```

Comment:
Template switch actually also has this wrong, but this should be `CONF_ENTITY_ID`. They end up the same functionally, but these should technically only be used within their proper domain to be correct.